### PR TITLE
WIP: Improve testing

### PR DESCRIPTION
**Pull request by shoekstra**
- Drop support/testing for CentOS 6, Fedora OS, FreeBSD
- Drop test suite for `allowed_ips` - no functional tests, no point
- Clean up kitchen config
- Test Chef 12 and Chef 13 but allow Chef 13 to fail until it's properly supported

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -35,6 +35,9 @@ suites:
     - recipe[nrpe::default]
     - recipe[nagios_test::default]
     - role[monitoring]
+    verifier:
+      inspec_tests:
+        - test/smoke/server_package
```
Comment (damacus) [inline .kitchen.yml]: You shouldn't need this line, unless I'm misunderstanding what you're trying to do?
https://github.com/sous-chefs/line-cookbook/tree/master/test/integration
https://github.com/sous-chefs/line-cookbook/blob/master/.kitchen.yml

File: .kitchen.yml [hunk 2]
```diff
@@ -1,96 +1,76 @@
 driver:
   name: vagrant
+  network:
+    - ["forwarded_port", {guest: 80, host: 8080}]
 
 provisioner:
   name: chef_zero
+  require_chef_omnibus: <%= ENV['CHEF_VERSION'] || 'latest' %>
```
Comment (damacus) [inline .kitchen.yml:8]: You should be able to control the Chef version like this iirc 

https://github.com/sous-chefs/haproxy/blob/master/.travis.yml#L26

File: .kitchen.yml [hunk 3]
```diff
@@ -1,96 +1,76 @@
...
+
+verifier:
+  name: inspec
 
 platforms:
   - name: ubuntu-14.04
-    driver_config:
-      network:
-        - ["forwarded_port", {guest: 80, host: 8081}]
```
Comment (damacus) [inline .kitchen.yml]: + 💯 for the de-duplication!

File: .kitchen.yml (hunk 2 above)
Comment (damacus) [reply .kitchen.yml:8]: https://github.com/sous-chefs/haproxy/blob/master/.kitchen.yml

^here's the haproxy config that @rshade mostly wrote.

File: .kitchen.yml (hunk 1 above)
Comment (shoekstra) [reply .kitchen.yml]: It's the recommended directory structure for inspec smoke tests when using test-kitchen is under `test/smoke/SUITE_NAME`, and specifying it like that is from the default chefdk generator cookbook so figured I'd keep that up.

File: .kitchen.yml (hunk 2 above)
Comment (shoekstra) [reply .kitchen.yml:8]: The above works fine, been using it internally for ages and is working for this project too.

**Comment by tas50** [comment]
I'm going to merge this and then nuke the opensuse testing for now. That' broken in Apache2 which is the root cause.

**Pull request merged**